### DISCUSSION TITLE
Modifying traffic identifier (TID) for video stream

### DISCUSSION THREAD

**Discussion by kuldeep-gill-skydio** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1564] [Q&A] [answered]
Does anyone know how I can modify the tid for udp packets sent using kinesis webrtc? Capturing the packets through wireshark, I see all the video packets go as besteffort and I would like to change it to video to reduce the contention time.

**Comment by MushMal** [comment]
If you are referring top DSCP/TOS on the packet? I don't think anything is being set and whether it actually will make any difference. However, you might want to try to set socket options here: 

https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/ced19d02ed6fbc1e59dd63bc19d847c5ec5ebf47/src/source/Sctp/Sctp.c#L17

and see whether it makes any difference in your ISP.

Would love to hear back

**Reply by kuldeep-gill-skydio** [reply]
Thanks! I will try and evaluate if it provides any latency benefit.

**Reply by kuldeep-gill-skydio** [reply]
@MushMal I tried adding the TOS here but the packets are still coming with best effort. Do I need to make changes elsewhere too?
```
@@ -22,6 +22,7 @@ STATUS configureSctpSocket(struct socket* socket)
     struct sctp_event event;
     UINT32 i;
     UINT32 valueOn = 1;
+    UINT32 tid_value = 0x80; // sending the data as video tid
     UINT16 event_types[] = {SCTP_ASSOC_CHANGE,   SCTP_PEER_ADDR_CHANGE,      SCTP_REMOTE_ERROR,
                             SCTP_SHUTDOWN_EVENT, SCTP_ADAPTATION_INDICATION, SCTP_PARTIAL_DELIVERY_EVENT};
 
@@ -36,6 +37,9 @@ STATUS configureSctpSocket(struct socket* socket)
     // delays are introduced, at the cost of more packets in the network.
     CHK(usrsctp_setsockopt(socket, IPPROTO_SCTP, SCTP_NODELAY, &valueOn, SIZEOF(valueOn)) == 0, STATUS_SCTP_SESSION_SETUP_FAILED);
 
+    // by default packets are sent as besteffort
+    // modifying the TOS to video to enhance performance in congested enviornment
+    CHK(usrsctp_setsockopt(socket, IPPROTO_IP, IP_TOS, &tid_value, SIZEOF(tid_value)) == 0, STATUS_SCTP_SESSION_SETUP_FAILED);   
     MEMSET(&event, 0, SIZEOF(event));
     event.se_assoc_id = SCTP_FUTURE_ASSOC;
     event.se_on = 1;
```

**Reply by kuldeep-gill-skydio** [reply]
This doesn't work either: 
```
CHK(usrsctp_setsockopt(socket, IPPROTO_SCTP, IP_TOS, &tid_value, SIZEOF(tid_value)) == 0, STATUS_SCTP_SESSION_SETUP_FAILED); 
```

**Comment by MushMal** [comment, marked as answer]
Oops, I am terribly sorry. I pointed to the location for sending/receiving the data channel bits and not the rtp.

Here is where you can configure and check whether your network equipment actually honors the dscp/tos

https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/08e4d294a7466f42033a7aaeacf2521f519749f7/src/source/Ice/Network.c#L168

You can issue another setsockopt call with IPPROTO_IP/IP_TOS or a more platform independent way using SOL_SOCKET/SO_PRIORITY
